--- /usr/share/jolla-messages/pages/conversation/SMSMessageDelegate.qml.orig	2018-11-07 21:28:27.500941947 +0100
+++ /usr/share/jolla-messages/pages/conversation/SMSMessageDelegate.qml	2018-11-07 23:10:36.866423881 +0100
@@ -102,12 +102,36 @@
     }
 
     LinkedText {
+        id: subjectText
+        height: (modelData.subject != "") ? implicitHeight : 0
+        anchors {
+            left: parent.left
+            right: parent.right
+            leftMargin: Theme.horizontalPageMargin
+            rightMargin: Theme.horizontalPageMargin
+        }
+        wrapMode: Text.Wrap
+
+        plainText: {
+             if (modelData.subject != "") {
+                hasText = true
+                return modelData.subject
+            }
+        }
+        color: (message.highlighted || !inbound) ? Theme.highlightColor : Theme.primaryColor
+        linkColor: inbound || message.highlighted ? Theme.highlightColor : Theme.primaryColor
+        font.pixelSize: inbound ? Theme.fontSizeMedium : Theme.fontSizeSmall
+        horizontalAlignment: inbound ? Qt.AlignRight : Qt.AlignLeft
+    }
+
+    LinkedText {
         id: messageText
         anchors {
             left: inbound ? parent.left : attachments.right
             right: inbound ? attachments.left : parent.right
             leftMargin: inbound ? sidePadding : (attachments.height ? Theme.paddingMedium : Theme.horizontalPageMargin)
             rightMargin: !inbound ? sidePadding : (attachments.height ? Theme.paddingMedium : Theme.horizontalPageMargin)
+            top: subjectText.bottom
         }
 
         property int sidePadding: Theme.itemSizeSmall + Theme.horizontalPageMargin
@@ -122,9 +146,6 @@
             } else if (modelData.freeText != "") {
                 hasText = true
                 return modelData.freeText
-            } else if (modelData.subject != "") {
-                hasText = true
-                return modelData.subject
             } else if (modelData.eventType === CommHistory.MMSEvent) {
                 hasText = false
                 //% "Multimedia message"
